### PR TITLE
Convert all config files to ES6 and remove unused controller

### PR DESCRIPTION
**Pull request by dkanada**

### CODE REVIEW

File: src/scripts/settings/userSettings.js [hunk 1]
```diff
@@ -39,15 +38,15 @@ define(['appSettings', 'events'], function (appSettings, events) {
         });
     };
 
-    UserSettings.prototype.getData = function () {
+    export function getData() {
         return this.displayPrefs;
     };
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:43]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 2]
```diff
@@ -39,15 +38,15 @@ define(['appSettings', 'events'], function (appSettings, events) {
...
 
-    UserSettings.prototype.importFrom = function (instance) {
+    export function importFrom(instance) {
         this.displayPrefs = instance.getData();
     };
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:47]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 3]
```diff
@@ -64,7 +63,7 @@ define(['appSettings', 'events'], function (appSettings, events) {
         return result;
     };
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:64]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 4]
```diff
@@ -73,7 +72,7 @@ define(['appSettings', 'events'], function (appSettings, events) {
         return appSettings.get(name, userId);
     };
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:73]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 5]
```diff
@@ -84,7 +83,7 @@ define(['appSettings', 'events'], function (appSettings, events) {
         });
     };
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:84]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 6]
```diff
@@ -93,7 +92,7 @@ define(['appSettings', 'events'], function (appSettings, events) {
         return val !== 'false';
     };
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:93]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 7]
```diff
@@ -102,7 +101,7 @@ define(['appSettings', 'events'], function (appSettings, events) {
...
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:102]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 8]
```diff
@@ -120,7 +119,7 @@ define(['appSettings', 'events'], function (appSettings, events) {
...
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:120]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 9]
```diff
@@ -129,7 +128,7 @@ define(['appSettings', 'events'], function (appSettings, events) {
...
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:129]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 10]
```diff
@@ -138,79 +137,79 @@ define(['appSettings', 'events'], function (appSettings, events) {
...
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:138]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 11]
```diff
@@ -138,79 +137,79 @@ define(['appSettings', 'events'], function (appSettings, events) {
...
 
-    UserSettings.prototype.language = function (val) {
+    export function language(val) {
         if (val != null) {
             return this.set('language', val.toString(), false);
         }
 
         return this.get('language', false);
     };
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:146]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 12]
```diff
@@ -138,79 +137,79 @@ define(['appSettings', 'events'], function (appSettings, events) {
...
 
-    UserSettings.prototype.dateTimeLocale = function (val) {
+    export function dateTimeLocale(val) {
         if (val != null) {
             return this.set('datetimelocale', val.toString(), false);
         }
 
         return this.get('datetimelocale', false);
     };
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:154]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 13]
```diff
@@ -138,79 +137,79 @@ define(['appSettings', 'events'], function (appSettings, events) {
...
 
-    UserSettings.prototype.skipBackLength = function (val) {
+    export function skipBackLength(val) {
         if (val != null) {
             return this.set('skipBackLength', val.toString());
         }
 
         return parseInt(this.get('skipBackLength') || '10000');
     };
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:162]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 14]
```diff
@@ -138,79 +137,79 @@ define(['appSettings', 'events'], function (appSettings, events) {
...
 
-    UserSettings.prototype.skipForwardLength = function (val) {
+    export function skipForwardLength(val) {
         if (val != null) {
             return this.set('skipForwardLength', val.toString());
         }
 
         return parseInt(this.get('skipForwardLength') || '30000');
     };
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:170]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 15]
```diff
@@ -138,79 +137,79 @@ define(['appSettings', 'events'], function (appSettings, events) {
...
 
-    UserSettings.prototype.dashboardTheme = function (val) {
+    export function dashboardTheme(val) {
         if (val != null) {
             return this.set('dashboardTheme', val);
         }
 
         return this.get('dashboardTheme');
     };
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:178]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 16]
```diff
@@ -138,79 +137,79 @@ define(['appSettings', 'events'], function (appSettings, events) {
...
 
-    UserSettings.prototype.skin = function (val) {
+    export function skin(val) {
         if (val != null) {
             return this.set('skin', val, false);
         }
 
         return this.get('skin', false);
     };
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:186]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 17]
```diff
@@ -138,79 +137,79 @@ define(['appSettings', 'events'], function (appSettings, events) {
...
 
-    UserSettings.prototype.theme = function (val) {
+    export function theme(val) {
         if (val != null) {
             return this.set('appTheme', val, false);
         }
 
         return this.get('appTheme', false);
     };
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:194]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 18]
```diff
@@ -138,79 +137,79 @@ define(['appSettings', 'events'], function (appSettings, events) {
...
 
-    UserSettings.prototype.screensaver = function (val) {
+    export function screensaver(val) {
         if (val != null) {
             return this.set('screensaver', val, false);
         }
 
         return this.get('screensaver', false);
     };
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:202]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 19]
```diff
@@ -138,79 +137,79 @@ define(['appSettings', 'events'], function (appSettings, events) {
...
 
-    UserSettings.prototype.soundEffects = function (val) {
+    export function soundEffects(val) {
         if (val != null) {
             return this.set('soundeffects', val, false);
         }
 
         return this.get('soundeffects', false);
     };
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:210]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 20]
```diff
@@ -220,7 +219,7 @@ define(['appSettings', 'events'], function (appSettings, events) {
         return query;
     };
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:220]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 21]
```diff
@@ -233,23 +232,20 @@ define(['appSettings', 'events'], function (appSettings, events) {
         return this.set(key, JSON.stringify(values));
     };
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:233]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 22]
```diff
@@ -233,23 +232,20 @@ define(['appSettings', 'events'], function (appSettings, events) {
...
 
-    UserSettings.prototype.getSubtitleAppearanceSettings = function (key) {
+    export function getSubtitleAppearanceSettings(key) {
         key = key || 'localplayersubtitleappearance3';
         return JSON.parse(this.get(key, false) || '{}');
     };
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:238]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 23]
```diff
@@ -233,23 +232,20 @@ define(['appSettings', 'events'], function (appSettings, events) {
...
 
-    UserSettings.prototype.setSubtitleAppearanceSettings = function (value, key) {
+    export function setSubtitleAppearanceSettings(value, key) {
         key = key || 'localplayersubtitleappearance3';
         return this.set(key, JSON.stringify(value), false);
     };
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:243]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 24]
```diff
@@ -233,23 +232,20 @@ define(['appSettings', 'events'], function (appSettings, events) {
...
 
-    UserSettings.prototype.setFilter = function (key, value) {
+    export function setFilter(key, value) {
         return this.set(key, value, true);
     };
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:247]:
```suggestion
    }
```

File: src/scripts/settings/userSettings.js [hunk 25]
```diff
@@ -233,23 +232,20 @@ define(['appSettings', 'events'], function (appSettings, events) {
...
 
-    UserSettings.prototype.getFilter = function (key) {
+    export function getFilter(key) {
         return this.get(key, true);
     };
```
Comment (MrTimscampi) [inline src/scripts/settings/userSettings.js:251]:
```suggestion
    }
```

File: src/scripts/settings/appSettings.js [hunk 26]
```diff
@@ -108,28 +107,40 @@ define(['appStorage', 'events'], function (appStorage, events) {
...
-    AppSettings.prototype.get = function (name, userId) {
+    export function get(name, userId) {
         return appStorage.getItem(getKey(name, userId));
-    };
+    }
 
-    return new AppSettings();
-});
+    export default {
+        enableAutoLogin: enableAutoLogin,
```
Comment (YouKnowBlom) [inline src/scripts/settings/appSettings.js:134]: Isn't the individual `export`s on the functions redundant since this does the same thing?

Comment (dkanada) [reply src/scripts/settings/appSettings.js:134]: I think @dmitrylyzo was adding them for compatibility reasons.

Comment (dmitrylyzo) [reply src/scripts/settings/appSettings.js:134]: Yes, this need for compatibility when you import ES6-rewritten module. Without `default`, I got error `default is undefined`, iirc
But I didn't try to remove individual exports leaving only default.

With individual and default exports, result will be:
```
{
    _esModule: true, <-- iirc
    func1: ...,
    func2: ...,
    default: {
        func1: ...,
        func2: ...
    }
}
```

Comment (YouKnowBlom) [reply src/scripts/settings/appSettings.js:134]: Ah alright, makes sense to me.

File: package.json [hunk 27]
```diff
@@ -91,10 +91,12 @@
           "src/components/input/keyboardnavigation.js",
           "src/components/scrollManager.js",
           "src/components/sanatizefilename.js",
+          "src/components/scrollManager.js",
           "src/scripts/settings/webSettings.js",
           "src/scripts/settings/appSettings.js",
           "src/scripts/settings/userSettings.js",
           "src/scripts/imagehelper.js"
```
Comment (dmitrylyzo) [inline package.json:98]:
```suggestion
          "src/scripts/imagehelper.js",
```

File: package.json [hunk 28]
```diff
@@ -91,10 +91,12 @@
           "src/components/input/keyboardnavigation.js",
           "src/components/scrollManager.js",
           "src/components/sanatizefilename.js",
+          "src/components/scrollManager.js",
```
Comment (dmitrylyzo) [inline package.json:94]:
```suggestion
```

**Comment by MrTimscampi** [comment]
Some more suggestions for this:

Replace
https://github.com/jellyfin/jellyfin-web/blob/es6/src/scripts/settings/appSettings.js#L8
With
```javascript
name = '${userId}-${name}';
```

Replace
https://github.com/jellyfin/jellyfin-web/blob/es6/src/scripts/settings/appSettings.js#L31
With
```javascript
var key = 'enableautobitratebitrate-${mediaType}-${isInNetwork}';
```

Replace
https://github.com/jellyfin/jellyfin-web/blob/es6/src/scripts/settings/webSettings.js#L4
With
```javascript
if (data) {
    return Promise.resolve(data);
}
```

Replace
https://github.com/jellyfin/jellyfin-web/blob/es6/src/scripts/settings/webSettings.js#L5
With
```javascript
return fetch('/config.json?nocache=${new Date().getUTCMilliseconds()}').then(function (response) {
```

**Comment by dkanada** [comment]
@MrTimscampi I really think string literals should be used sparingly in any project. Your first two seem fine but I'm not a fan of anything past variables between the brackets. Also, I think inline if statements are also fine when used sparingly.

**Pull request merged**